### ISSUE TITLE
CI: let TeamCity track the tests in nightly stress

### ISSUE THREAD

**Issue by tbg** [cockroachdb/cockroach#67258]
The [TC nightly stress](https://teamcity.cockroachdb.com/admin/editBuildFeatures.html?id=buildType:Cockroach_Nightlies_Stress) jobs isn't configured to parse the `go test -json` output yet. If we do this, the stress runs will both feed into the "Flaky test" pipeline (which is helpful) and we will be able to see, on TeamCity, which tests are slow (for a given package). We need to do a few things here:

1. add the golang build feature
2. teach `stress` to stream stdout
3. invoke `stress` (roughly) with `stress go test ./pkg/util/stop -json -exec './stop.test -test.v'` instead of `stress ./stop.test` (to enable json output) in `build/teamcity-stress.sh`

**Comment by tbg** [comment]
@rickystewart I'm curious how this would work in bazel. Right now, we have a nightly job that triggers one of the above builds for each package. The package runs for usually 1h or 100 iterations, whichever is shorter.

We've found this isn't really that great (packages have very different numbers of tests), we'd rather have this on a per-test basis (i.e. run each test for 3 minutes or 1000 iterations, whichever happens faster). Any idea what this would look like in a bazel world? Would the basic version of this be as easy as `bazel test ./pkg/... --runs_per_test=1000`? That doesn't have the "or 3 minutes" bit but maybe that's fine?

**Comment by rickystewart** [comment]
@tbg 

> Would the basic version of this be as easy as bazel test ./pkg/... --runs_per_test=1000? That doesn't have the "or 3 minutes" bit but maybe that's fine?

Yes. This is the plan now that we have access to a remote execution cluster. 1000 is probably too high a figure; we'll have to experiment to find a figure that stresses "enough". You're right that we don't have access to the "or 3 minutes" bit of the logic, but we can forego that.
